design/authorization-handler.md

```diff
@@ -0,0 +1,590 @@
+
+A static OAuth scope such as `address` will be parsed similar to the previous cases, but with the `identifier` field missing:
+```json
+{
+	"type": "address"
+}
+```
+
+The client scope will be determined again from the type field.
+
+The request will be rejected in case that there are any unknown scope values or unknown authorization_detail types, which don't correspond to any configured authorization handlers in the realm. The error returned to the client is `invalid_authorization_details` as described in the [RAR specification Section 8](https://datatracker.ietf.org/doc/html/draft-ietf-oauth-rar-07#section-8) or `invalid_scope` as it works today and described in the OAuth2 specification.
+
+### AuthorizationContext in AuthenticationSessionModel
+
+The AuthorizationContext can be sent as a field in the AuthenticationSessionModel. Hence it will be available in all the requests of this Authentication session. Existing AuthenticationSession methods “get/setClientScopes” may be deprecated and replaced by “get/setAuthenticationContext”
+
+## Compliance processor
+
+There will be a chain of policies configured in the admin console for a specific Authorization  Handler. The AuthorizationContext can be processed by this policy chain. Each AuthorizationDetailsEntry will be processed by each of the policy and policy will decide if:
+
+* AuthorizationDetails is required to be rendered on the consent screen
+* AuthorizationDetails is unsupported/invalid/unauthorized and whole client request should be rejected
+
+Policy can also “enrich” and/or change authorizationDetails entries as sometimes client sent “incomplete” authorization_details and it is up to the AS (Keycloak server) to figure correct values and then in later stages, display them to the user on consent screen or add them to the token. Some use-case described in the [RAR specification Section 7.1](https://datatracker.ietf.org/doc/html/draft-ietf-oauth-rar-07#section-7.1).
+
+Policy is also responsible for processing the consent screen (The moment when a user submits the consent screen, which can happen multiple times during a single “authentication session” due the fact that more consent screens would be possible with this proposal).
+
+### Naming
+
+Maybe ClientScopesPolicy is ok as a name? IMO any name containing "Authorization" (EG. AuthorizationContextPolicy, AuthorizationDetailsPolicy) can bring the confusion with the authorization services, which also use "Policies"... More wider question if AuthorizationContext itself is good as a name?
+
+### Configuration
+
+Policies can be configured at the client scope (or AuthorizationHandler) level. In addition to tab Mappers and Scope, there can be another tab like Policies . By default, each client scope can have only Persistent Consent Policy automatically set (See below for Persistent Consent Policy details).
+
+### When
```

Comment:
It might be better to check `AuthorizationDetails` is acceptable or not at the beginning of the request to AuthorizationEndpoint.

Reply:
Added it in the `### ClientAuthorizationContext initiator` section to make sure it's accounted for.